### ISSUE TITLE
gesture: infinite recursion in GestureHandler.end() with Android TalkBack

### ISSUE THREAD

**Issue by vbraun** [angular/material#11768]
## Bug

Clicking on a button with android talkback causes an infinite recusion when dispatching the event. Moreover, this ends up delivering the native event again after the exception.

The duplicate click event causes major breakage with talkback. For example, you can't open the sidebar menu dropdowns on https://material.angularjs.org as they open and then immediately close.

#### Detailed Reproduction Steps: 
1. Open https://material.angularjs.org/latest/demo/button on android with talkback enabled
1. Attach remote debugger
1. Click button

### What is the expected behavior?

Button should be clicked once without js error

### What is the current behavior?

````
Uncaught RangeError: Maximum call stack size exceeded.
    at GestureHandler.nativeDispatchEvent [as dispatchEvent] (angular-material.js:4131)
    at GestureHandler.onEnd (angular-material.js:3730)
    at GestureHandler.end (angular-material.js:4000)
    at runHandlers (angular-material.js:4241)
    at HTMLDocument.gestureEnd (angular-material.js:4292)
    at defaultHandlerWrapper (angular.js:3802)
    at HTMLDocument.eventHandler (angular.js:3790)
    at GestureHandler.nativeDispatchEvent [as dispatchEvent] (angular-material.js:4131)
    at GestureHandler.onEnd (angular-material.js:3730)
    at GestureHandler.end (angular-material.js:4000)
````
(note: this is the non-minified stacktrace for clarity)

### Which versions of AngularJS, Material, OS, and browsers are affected?

Angularjs material 1.1.19
Android 9 (Pixel 3 XL)

**Comment by Splaktar** [comment]
I've been able to reproduce this back to `1.1.5` and it's likely to have been an issue since even earlier.

**Comment by Splaktar** [comment]
I've identified that this is caused by the final `mouseup` event re-firing infinite times due to click hijacking being enabled.
```js
dispatchEvent:  mousedown
dispatchEvent:  $md.pressdown
dispatchEvent:  click
dispatchEvent:  focus
dispatchEvent:  mouseup
dispatchEvent:  $md.pressup
dispatchEvent:  mouseup x 48 -> Uncaught RangeError: Maximum call stack size exceeded.
```
If click hijacking is disabled via `$mdGestureProvider.skipClickHijack();`, then this does not occur and the only events re-fired from a button click are:
```js
dispatchEvent:  $md.pressdown
dispatchEvent:  $md.pressup
```
I also have a fix for this that I'm still verifying...

**Comment by Splaktar** [comment]
PR https://github.com/angular/material/pull/11774 posted to fix this.
